f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py

```diff
@@ -500,9 +500,82 @@ def __init__(self, driver):
+        members_payload = []
+        name_list = []
+
+        """ old members from bigip """
+        for item in bigip_members:
+            fqdn = item.fqdn
+            content = item.to_dict()
+            del content['_meta_data']
+            del content['fqdn']
+            """ after to_dict the content's fqdn becomes a list somehow"""
+            content['fqdn'] = fqdn
+
+            if content['session'] == 'monitor-enabled':
+                del content['session']
+                del content['state']
+            elif content['session'] == 'user-disabled':
+                if content['state'] != 'user-down':
+                    del content['state']
+            else:
+                LOG.debug("Default case %s %s",
+                          content['session'], content['state'])
+                del content['session']
+                del content['state']
+
+            members_payload.append(content)
+            name_list.append(content['name'])
+
+        """ new members from bigip """
+        for item in lbaas_members:
+            if item['name'] not in name_list:
+                content = self._create_payload(item, service)
+                members_payload.append(content)
+
+        return members_payload
+
     @log_helpers.log_method_call
     def create(self, resource, service, **kwargs):
 
+        self.driver.prepare_network_for_member(service)
+        LOG.debug("Begin to create in batch %s %s",
+                  self._resource, resource['name'])
+        if not service.get(self._key):
+            self._search_element(resource, service)
+        member = service.get('member')
+        pool = {}
+        pool['id'] = member['pool_id']
+        self._pool_mgr._search_element(pool, service)
+        pool_payload = self._pool_mgr._create_payload(pool, service)
+
+        if 'members' not in pool_payload:
+            LOG.error("None members in pool")
+            return
+        lbaas_members = pool_payload['members']
+        bigips = self.driver.get_config_bigips()
+        for bigip in bigips:
+            pool_resource = self._pool_mgr.pool_helper.load(
+                bigip,
+                name=urllib.quote(pool_payload['name']),
+                partition=pool_payload['partition']
+            )
+            bigip_members = pool_resource.members_s.get_collection()
+
+            del pool_payload['members']
+            new_payload = self._merge_members(
+                lbaas_members, bigip_members, service)
+            pool_payload['members'] = new_payload
+            self._shrink_payload(pool_payload,
+                                 keys_to_keep=['partition', 'name', 'members'])
+            self._pool_mgr._update(bigip, pool_payload, None, None, service)
+        LOG.debug("Finish to create in batch %s %s",
+                  self._resource, resource['name'])
+
+    @log_helpers.log_method_call
+    def create_single(self, resource, service, **kwargs):
```

Comment:
when is this method used?  will you modify driver side code later as well (to use this create_single) ?  Maybe  we can just use the same create() for both the 2 scenarios: creating 1 or many members.

Reply:
yes, xiejun.  Currently, "create "method is used both for single and multiple scenarios.  Create_single isn't used anymore.

While we still want to optimize further that is, create is used for multiple-members and create_single is used for single members respectively. 

To do that, we need your help to add some logic in driver to classify these two scenarios in driver.   For example, add a k-v pair "multiple: True"   for multiple scenario and "multiple:false" for single scenario. So that in the agent code, when create_member is triggered, it could distinguish these two cases by the value of "the multiple" key in service.

Let's discuss this in detail if needed.

Thanks